library/Translate/test/Test/Contextify.hs

```diff
@@ -44,6 +44,22 @@ moduleResolution =
                 "(:lambda-case \
                 \  (() (case TopLevel.A.a ((Cons a (Cons x y)) (add a x y)))))"
         Just expected T.@=? unwrapLookup "fi" t,
+      T.testCase "infix declaration works with declaration" $ do
+        Right t <-
+          contextualizeFoo
+            "type foo = (::) declare infixl (::) 20 let fi = a * 3 :: 10"
```

Comment:
How do you know that `(::)` takes any arguments here?

Reply:
This is not being tested end to end

Reply:
Doesn't the LLVM only return int types, and since we can't pattern match, what is there to test with this?

Reply:
In this case we don't know the arity of `::`. In fact, if anything, this example would suggest that arity is 0, so infixivity does not make sense.

I would expect and end to end test. We don't need pattern matching.

Reply:
it says the arity

```haskell
infixl (::) 20 
```

I have another example where I set it to 1, to make sure the infixivity works

Reply:
As for coherence, this PR changes nothing about the coherence of infixivity constructors, it simply allows the `declare` to work over them. If they worked before this, then they worked before this, however this only changes the declaration thereof as what was wanted

Reply:
We want features to work. In this case, we want infixivity for sum constructors to work. What is the value from the user's perspective of infixivity to work partially (only for the parser)?